### PR TITLE
Enforcing use of SP's login domain in IDP app

### PR DESCRIPTION
**Pull request by trooper2013**
This PR covers the following. 
1. Enforcing use of domains & filtering of users related to domain in the IDP app.
2. Missing parameters like brand and addl values needed to be passed to the Coordinator for regular SFDC OAuth authentication features.
3. Ensure Error's are  displayed on main thread.

### CODE REVIEW

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 55%</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SalesforceRestAPITests | testBlockUpdate|
SalesforceRestAPITests | testBlockUpdate|
SalesforceRestAPITests | testUpsert|
SalesforceRestAPITests | testUpsert|


## Current coverage for SalesforceAnalytics.framework is `44.08%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `48.22%`
Files changed | - | - 
--- | --- | ---
SFUserAccountManager+URLHandlers.m | `0.00%` | :skull:
SFSDKUserSelectionTableViewController.m | `0.00%` | :skull:
SFSDKUserSelectionNavViewController.m | `0.00%` | :skull:
SFSDKIDPAuthClient.m | `1.29%` | :skull:
SFSDKOAuthClient.m | `31.69%` | :no_entry_sign:
SFUserAccountManager.m | `59.05%` | :warning:
SFSDKIDPInitCommand.m | `60.00%` | :warning:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKIDPAuthClient.m [hunk 1]
```diff
@@ -106,6 +106,7 @@ - (BOOL)refreshCredentials {
 }
 
 - (BOOL)initiateLocalLoginInSPApp {
+    self.config.hideSettingsIcon = NO;
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKIDPAuthClient.m]: Is that true? Should it always be shown in the SP app? Shouldn't we go with whatever the app wants?

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorerUITests/OAuthUITests.swift [hunk 2]
```diff
@@ -109,7 +109,7 @@ class OAuthUITest: SalesforceNoSessionTestCase {
             sleep(1)
         }
         
-        SFAuthenticationManager.shared().logoutAllUsers() //FIXME: for some reason, this doesn's work
+        SFUserAccountManager.sharedInstance().logoutAllUsers() //FIXME: for some reason, this doesn's work
```
Comment (wmathurin) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorerUITests/OAuthUITests.swift]: What that FIXME about?

Comment (trooper2013) [reply native/SampleApps/SmartSyncExplorer/SmartSyncExplorerUITests/OAuthUITests.swift]: @wmathurin  IMHO the future of these UItests are questionable. We don't use them. I only moved SFAuthenticationManager to SFUserAccountManager. The "Fixme" reason was not clear in the first place. I left it the way it was. We need to talk about the status of UI tests in the SDK. They aren't being used for testing.  Should we continue to keep them?

Comment (wmathurin) [reply native/SampleApps/SmartSyncExplorer/SmartSyncExplorerUITests/OAuthUITests.swift]: (Outside this PR of course) we should have unit tests for a sample apps - especially for things that cannot be completely tested in unit tests e.g logout / login.

**Pull request merged**